File: LATIHAN1DPBO2023/Python/CRUD.py
# Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 1
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.

#import class
from Mahasiswa import Mahasiswa

#deklarasi class CRUD
class CRUD:
	#atribut untuk tampungan
	name = ""
	nim = ""
	major = ""
	faculty = ""

	#constructor
	def __init__():
		i = 0 #katanya gak boleh kosong

	#method untuk menambahkan data (create)
	def add(mmahasiswa):
		j = 0

		#inputan untuk data mahasiswa
		print("Masukkan Nama:")
		name = str(input())
		print("Masukkan NIM:")
		nim = str(input())
		print("Masukkan Program Studi:")
		major = str(input())
		print("Masukkan Fakultas:")
		faculty = str(input())

		#pengecekan apakah data sudah tersedia pada list (berdasarkan nim)
		for mahasiswa in mmahasiswa:
			if(nim == mahasiswa.getNim()):
				j = 1
				print("\nTidak dapat menambahkan data, NIM sudah tersedia!")

		#jika tidak tersedia maha masukkan ke list
		if(j == 0):
			mmahasiswa.append(Mahasiswa(name, nim, major, faculty))
			print("\nData berhasil ditambahkan!")

	#method untuk mengubah data (update)
	def update(nim, mmahasiswa):
		j = 0

		#perulangan untuk mencari data yang akan diubah (berdasarkan nim)
		for mahasiswa in mmahasiswa:
			if(nim == mahasiswa.getNim()):
				#inputan untuk data yang akan diubah
				print("Masukkan Nama:")
				name = str(input())
				print("Masukkan NIM:")
				nim = str(input())
				print("Masukkan Program Studi:")
				major = str(input())
				print("Masukkan Fakultas:")
				faculty = str(input())

				mahasiswa.setName(name);
				mahasiswa.setNim(nim);
				mahasiswa.setMajor(major);
				mahasiswa.setFaculty(faculty);

				print("\nData berhasil diubah!")

				j = 1

		#jika data tidak tersedia
		if(j == 0):
			print("\nData tidak tersedia!")

	#method untuk menghapus data (delete)
	def delete(nim, mmahasiswa):
		i = 0
		j = 0

		#perulangan untuk mencari data yang akan dihapus (berdasarkan nim)
		for mahasiswa in mmahasiswa:
			if(nim == mahasiswa.getNim()):
				del mmahasiswa[i]
				print("\nData berhasil dihapus!")
				j = 1
			i += 1

		#jika data tidak tersedia
		if(j == 0):
			print("\nData tidak tersedia!")

	#method untuk menampilkan data (read)
	def show(mmahasiswa):
		#output berupa tabel
		i = 0
		print("\nList Mahasiswa:")
		print("+=======================================+\n" + "|  Nama  |  NIM  |  Prodi  |  Fakultas  |\n" + "+=======================================+")
		for mahasiswa in mmahasiswa:
			print("| " + str(i + 1) + ". " + mahasiswa.getName() + " | " + mahasiswa.getNim() + " | " + mahasiswa.getMajor() + " | " + mahasiswa.getFaculty() + " |")
			print("+---------------------------------------+")
			i += 1

File: LATIHAN1DPBO2023/Java/CRUD.java
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//library yang digunakan
import java.util.Scanner;
import java.util.ArrayList;

//inisialisasi class CRUD
public class CRUD
{
	//atribut sebagai tempat tampung
	String name;
	String nim;
	String major;
	String faculty;
	int i, j;

	//membuat scanner untuk inputan
	Scanner sc = new Scanner(System.in);

	//constructor kosong
	public CRUD()
	{

	}

	//method untuk untuk menambahkan data (create)
	public void add(ArrayList<Mahasiswa> list)
	{
		//instance object menggunakan class Mahasiswa
		Mahasiswa temp = new Mahasiswa();
		j = 0;

		//inputan data mahasiswa
		System.out.print("Masukkan Nama:");
		name = sc.next();
		System.out.print("Masukkan NIM:");
		nim = sc.next();
		System.out.print("Masukkan Program Studi:");
		major = sc.next();
		System.out.print("Masukkan Fakultas:");
		faculty = sc.next();

		//perulangan untuk pengecekan apakah data sudah tersedia (berdasarkan nim)
		for(i = 0; i < list.size(); i++)
		{
			if(nim.equals(list.get(i).getNim()))
			{
				System.out.println("\nTidak dapat menambahkan data, NIM sudah tersedia!");
				j = 1;	
			}
		}

		//jika data belum tersedia maka set data dan masukkan ke list
		if(j == 0)
		{
			temp.setName(name);
			temp.setNim(nim);
			temp.setMajor(major);
			temp.setFaculty(faculty);

			list.add(temp);
			System.out.println("\nData berhasil ditambahkan!");
		}
	}

	//method untuk mengubah data (update)
	public void update(String nim, ArrayList<Mahasiswa> list)
	{
		j = 0;

		//perulangan untuk mencari data mahasiswa yang akan diubah (berdasarkan nim)
		for(i = 0; i < list.size(); i++)
		{
			//jika ketemu maka ubah data dengan inputan baru
			if(nim.equals(list.get(i).getNim()))
			{
				System.out.print("Masukkan Nama:");
				name = sc.next();
				System.out.print("Masukkan NIM:");
				nim = sc.next();
				System.out.print("Masukkan Program Studi:");
				major = sc.next();
				System.out.print("Masukkan Fakultas:");
				faculty = sc.next();

				list.get(i).setName(name);
				list.get(i).setNim(nim);
				list.get(i).setMajor(major);
				list.get(i).setFaculty(faculty);

				System.out.println("\nData berhasil diubah!");
				j = 1;
			}
		}

		//jika data tidak tersedia
		if(j == 0)
		{
			System.out.println("\nData tidak tersedia!");
		}
	}

	//method untuk menghapus data (delete)
	public void delete(String nim, ArrayList<Mahasiswa> list)
	{
		j = 0;

		//perulangan untuk mencari data mahasiswa yang akan dihapus (berdasarkan nim)
		for(i = 0; i < list.size(); i++)
		{
			//jika ketemu maka hapus data
			if(nim.equals(list.get(i).getNim()))
			{
				list.remove(list.get(i));
				System.out.println("\nData berhasil dihapus!");
				j = 1;
			}
		}

		//jika data tidak tersedia
		if(j == 0)
		{
			System.out.println("\nData tidak tersedia!");
		}
	}

	//method untuk menampilkan data (read)
	public void show(ArrayList<Mahasiswa> list)
	{
		//output berupa tabel
		System.out.println("\nList Mahasiswa:");
		System.out.println("+=======================================+\n" + "|  Nama  |  NIM  |  Prodi  |  Fakultas  |\n" + "+=======================================+");
		for(i = 0; i < list.size(); i++)
		{
			System.out.println("| " + Integer.toString(i + 1) + ". " + list.get(i).getName() + " | " + list.get(i).getNim() + " | " + list.get(i).getMajor() + " | " + list.get(i).getFaculty() + " |");
			System.out.println("+---------------------------------------+");
		}
	}
}